i18n/i18n.go

```diff
@@ -15,18 +15,20 @@
 
 package i18n
 
-import "github.com/arduino/arduino-cli/configuration"
-
 // Init initializes the i18n module, setting the locale according to this order of preference:
-// 1. Configuration set in arduino-cli.yaml
+// 1. Locale specified via the function call
 // 2. OS Locale
 // 3. en (default)
-func Init() {
+func Init(configLocale ...string) {
```

Comment:
Why do we need multiple locale strings if we use only one? :thinking:

Reply:
The reason I used a variadic parameter here is because the package has the fallback behavior of auto-detecting the user's locale in the event a preference was not specified.

I felt like an API that provided for that behavior like this:
```go
i18n.Init()
```
was more intuitive than one that required the user to do this:
```go
i18n.Init("")
```
I have a check that prevents multiple locale strings. You can pass zero strings for auto-detect, or one string to set the locale explicitly. Any more than that results in a panic.

I'm definitely not set on this approach. I had it the other way originally, but switched to the variadic approach at the last minute because the empty string approach felt awkward to me while I was explaining it in the `UPGRADING.md` entry.

Reply:
Makes sense, never thought about using variadic params like this really but I like the creativity. Approving. :+1: